lib/core/src/lib/testing/unit-testing-utils.ts

```diff
@@ -0,0 +1,477 @@
+    }
+
+    mouseLeaveByDataAutomationId(dataAutomationId: string): void {
+        const element = this.getByDataAutomationId(dataAutomationId);
+        element.triggerEventHandler('mouseleave', new MouseEvent('mouseleave'));
+    }
+
+    keyBoardEventByCSS(selector: string, event: 'keyup' | 'keydown', code: string, key: string): void {
+        const element = this.getByCSS(selector);
+        element.nativeElement.dispatchEvent(new KeyboardEvent(event, { code, key }));
+    }
+
+    dispatchCustomEventByCSS(selector: string, eventName: string): void {
+        const element = this.getByCSS(selector);
+        element.nativeElement.dispatchEvent(new CustomEvent(eventName));
+    }
+
+    /** Input related methods */
+
+    getInputByCSS(selector: string): HTMLInputElement {
+        return this.getByCSS(selector)?.nativeElement;
+    }
+
+    getInputByDataAutomationId(dataAutomationId: string): HTMLInputElement {
+        return this.getByDataAutomationId(dataAutomationId)?.nativeElement;
+    }
+
+    fillInputByCSS(selector: string, value: string): void {
+        const input = this.getInputByCSS(selector);
+        input.value = value;
+        input.dispatchEvent(new Event('input'));
+    }
+
+    fillInputByDataAutomationId(dataAutomationId: string, value: string): void {
+        const input = this.getInputByDataAutomationId(dataAutomationId);
+        input.value = value;
+        input.dispatchEvent(new Event('input'));
+    }
+
+    /** MatButton related methods */
+
+    async getMatButton(): Promise<MatButtonHarness> {
+        return this.loader.getHarness(MatButtonHarness);
+    }
+
+    async getMatButtonByCSS(selector: string): Promise<MatButtonHarness> {
+        return this.loader.getHarness(MatButtonHarness.with({ selector }));
+    }
+
+    async getMatButtonByDataAutomationId(dataAutomationId: string): Promise<MatButtonHarness> {
+        return this.loader.getHarness(MatButtonHarness.with({ selector: `[data-automation-id="${dataAutomationId}"]` }));
+    }
+
+    async checkIfMatButtonExists(): Promise<boolean> {
+        return this.loader.hasHarness(MatButtonHarness);
+    }
+
+    async checkIfMatButtonExistsWithDataAutomationId(dataAutomationId: string): Promise<boolean> {
+        return this.loader.hasHarness(MatButtonHarness.with({ selector: `[data-automation-id="${dataAutomationId}"]` }));
+    }
+
+    async clickMatButton(): Promise<void> {
+        const button = await this.getMatButton();
+        await button.click();
+    }
+
+    async clickMatButtonByCSS(selector: string): Promise<void> {
+        const button = await this.getMatButtonByCSS(selector);
+        await button.click();
+    }
+
+    async clickMatButtonByDataAutomationId(dataAutomationId: string): Promise<void> {
+        const button = await this.getMatButtonByDataAutomationId(dataAutomationId);
+        await button.click();
+    }
+
+    async sendKeysToMatButton(keys: string[] | TestKey[]): Promise<void> {
```

Comment:
we can apply the same comment like https://github.com/Alfresco/alfresco-ng2-components/pull/10571/files#r1931893317 here

Reply:
Done